lib/vmdb/settings/database_source.rb

```diff
@@ -3,23 +3,51 @@ class Settings
     class DatabaseSource
       include Vmdb::Logging
 
-      attr_reader :resource
+      attr_reader :resource, :settings_change_holder
 
-      def initialize(resource)
-        @resource = resource
+      def initialize(resource_to_get_settings_about, class_to_call_settings_changes_on)
+        @resource = resource_to_get_settings_about
+        @settings_change_holder = class_to_call_settings_changes_on
```

Comment:
I did it in first draft, but did not like myself. Looking at the code now, I think it will be clear.